assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -56,8 +56,17 @@ function MediaProvider({ children }) {
   const {
     actions: { getMedia },
   } = useAPI();
+
+  // Used by callbacks to avoid unnecessary dependency on state changes,
+  // which cause re-renderings.
+  const stateRef = useRef();
+  stateRef.current = state;
```

Comment:
```suggestion
```

Reply:
The state must be updated on each re-render. Passing 'state' to useRef() only assigns an initial value.

Reply:
Resolving, re-open if there is still a concern